### PR TITLE
fix ‘SWP01 - Use of Assert instead of throw’

### PR DESCRIPTION
**Pull request by vikkkko**

### CODE REVIEW

File: Swap/flamingo-contract-swap/FlamingoSwapFactory/FlamingoSwapFactoryContract.Admin.cs [hunk 1]
```diff
@@ -57,11 +57,13 @@ public static bool SetAdmin(UInt160 admin)
         /// <param name="data"></param>
         public static void Update(ByteString nefFile, string manifest, object data)
         {
-            if (!Verify()) throw new Exception("No authorization.");
+            if (!Verify())
```
Comment (Ashuaidehao) [inline Swap/flamingo-contract-swap/FlamingoSwapFactory/FlamingoSwapFactoryContract.Admin.cs:60]:
```suggestion
            Assert(Verify(), "No authorization.");

```

File: Swap/flamingo-contract-swap/FlamingoSwapFactory/FlamingoSwapFactoryContract.Helper.cs [hunk 2]
```diff
@@ -15,10 +15,7 @@ public partial class FlamingoSwapFactoryContract
         /// <param name="message"></param>
         private static void Assert(bool condition, string message)
         {
-            if (!condition)
-            {
-                throw new Exception(message);
-            }
+            Assert(condition, message);
```
Comment (Ashuaidehao) [inline Swap/flamingo-contract-swap/FlamingoSwapFactory/FlamingoSwapFactoryContract.Helper.cs:18]: Call self?

File: Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Nep17.cs [hunk 3]
```diff
@@ -15,14 +16,14 @@ partial class FlamingoSwapPairContract
 
         public static bool Transfer(UInt160 from, UInt160 to, BigInteger amount, object data)
         {
-            if (amount <= 0) throw new Exception("The parameter amount MUST be greater than 0.");
-            if (!Runtime.CheckWitness(from) && !from.Equals(Runtime.CallingScriptHash)) throw new Exception("No authorization.");
+            Assert(amount > 0, "The parameter amount MUST be greater than 0.");
+            Assert(Runtime.CheckWitness(from) || from.Equals(Runtime.CallingScriptHash), "No authorization.");
```
Comment (Ashuaidehao) [inline Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Nep17.cs:20]: No need check `from.Equals(Runtime.CallingScriptHash)` in N3.

File: Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs [hunk 4]
```diff
@@ -50,7 +51,7 @@ private static void Assert(bool condition, string message, params object[] data)
         /// <returns></returns>
         private static BigInteger Sqrt(BigInteger y)
```
Comment (shargon) [inline Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs:52]: Remove `Sqrt` method?

**Review by shargon — APPROVED** [review]

File: Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs (hunk 4 above)
Comment (vikkkko) [reply Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs:52]: ok

Comment (shargon) [reply Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs:52]: But there is a native one in BigInteger that use the opcode SQRT
https://github.com/neo-project/neo-devpack-dotnet/blob/4589948d84d2127b617406ea7a316fcf5f6b7cff/src/Neo.SmartContract.Framework/Helper.cs#L222

Comment (vikkkko) [reply Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs:52]: I'm going to use the native method

File: Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs [hunk 5]
```diff
@@ -23,7 +23,8 @@ private static void Assert(bool condition, string message)
         {
```
Comment (shargon) [inline Swap/flamingo-contract-swap/FlamingoSwapPair/FlamingoSwapPairContract.Helper.cs:23]: You can do the same as FLM, remove this method, and use only the native one

**Pull request merged**